nifi-nar-bundles/nifi-windows-event-log-bundle/nifi-windows-event-log-processors/src/main/java/org/apache/nifi/processors/windows/event/log/EvtSubscribe.java

```diff
@@ -0,0 +1,272 @@
+            .description("Maximum number of events to queue for transformation into FlowFiles before the Processor starts running.")
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .build();
+
+    public static final List<PropertyDescriptor> PROPERTY_DESCRIPTORS = Collections.unmodifiableList(Arrays.asList(CHANNEL, QUERY, MAX_BUFFER_SIZE, MAX_EVENT_QUEUE_SIZE));
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("Relationship for successfully formatted events.")
+            .build();
+
+    public static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(REL_SUCCESS)));
+
+    private final AtomicReference<ProcessSessionFactory> sessionFactoryReference;
+    private final Queue<String> renderedXMLs;
+    private final WEvtApi wEvtApi;
+    private final Kernel32 kernel32;
+
+    private Throwable wEvtApiError = null;
+    private Throwable kernel32Error = null;
+
+    private WEvtApi.EVT_SUBSCRIBE_CALLBACK evtSubscribeCallback;
+    private WinNT.HANDLE subscriptionHandle;
+
+    private WEvtApi loadWEvtApi() {
+        try {
+            return WEvtApi.INSTANCE;
+        } catch (Throwable e) {
+            wEvtApiError = e;
+            return null;
+        }
+    }
+
+    private Kernel32 loadKernel32() {
+        try {
+            return Kernel32.INSTANCE;
+        } catch (Throwable e) {
+            kernel32Error = e;
+            return null;
+        }
+    }
+
+    /**
+     * Framework constructor
+     */
+    public EvtSubscribe() {
+        this(null, null);
+    }
+
+    /**
+     * Constructor that allows injection of JNA interfaces
+     *
+     * @param wEvtApi event api interface
+     * @param kernel32 kernel interface
+     */
+    public EvtSubscribe(WEvtApi wEvtApi, Kernel32 kernel32) {
+        this.wEvtApi = wEvtApi == null ? loadWEvtApi() : wEvtApi;
+        this.kernel32 = kernel32 == null ? loadKernel32() : kernel32;
+        this.sessionFactoryReference = new AtomicReference<>();
+        this.renderedXMLs = new LinkedList<>();
+    }
+
+    /**
+     * Register subscriber via native call
+     *
+     * @param context the process context
+     */
+    @OnScheduled
+    public void subscribeToEvents(ProcessContext context) {
+        int maxEventQueueSize = context.getProperty(MAX_EVENT_QUEUE_SIZE).asInteger();
+        evtSubscribeCallback = new EventSubscribeXmlRenderingCallback(getLogger(), s -> {
+            ProcessSessionFactory processSessionFactory = sessionFactoryReference.get();
+            if (processSessionFactory == null) {
+                addRenderedXml(s, maxEventQueueSize);
+            } else {
+                ProcessSession session = processSessionFactory.createSession();
+                createAndTransferEventFlowFile(session, s);
+                session.commit();
```

Comment:
In general you should refrain from transferring FlowFiles outside of an onTrigger. This processor is following the ConsumeX paradigm. Where there is a resource pushing data to the processor which is a bit tricky when a processor should only emit FlowFiles when triggered. 

For a good reference for how to address concerns like to store the data, how much to store, when to transferred the data and what to do when the processor is stopped (I transfer the lingering data to avoid data loss) check out my ConsumeMqtt processor[1]. It is the same general idea but instead of getting Windows events pushed to it, it gets MQTT messages pushed to it.

[1] https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-mqtt-bundle/nifi-mqtt-processors/src/main/java/org/apache/nifi/processors/mqtt/ConsumeMQTT.java